### PR TITLE
Relax the type checking on arguments to subroutine

### PR DESCRIPTION
**Pull request by barnjamin**
Addresses #530

### CODE REVIEW

File: pyteal/ast/subroutine.py [hunk 1]
```diff
@@ -282,7 +282,17 @@ def invoke(
                     raise TealInputError(
                         f"supplied argument at index {i} should be an ABI type but got {arg}"
                     )
-                if arg.type_spec() != arg_type:
+
+                # If the expected arg is a TransactionType we can relax our check
+                # to be any subclass of TransactionTypeSpec
+                if arg_type.__class__ is abi.TransactionTypeSpec:
```
Comment (ahangsu) [inline pyteal/ast/subroutine.py:288]: nit: would prefer code style to be
```suggestion
                if type(arg_type) is abi.TransactionTypeSpec:
```

**Review by ahangsu — APPROVED** [review]
generally looks good to me, one nit

**Comment by jasonpaulos** [comment]
leaving a note that I intend to review this soon

**Comment by jasonpaulos** [comment]
This is blocked by #540

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 * Added `abi.type_spec_is_assignable_to` to check for compatible ABI type assignments. ([#540](https://github.com/algorand/pyteal/pull/540))
 
 ## Fixed
+* Subroutines that take ABI type of Transaction now allow any Transaction type to be passed. ([#531](https://github.com/algorand/pyteal/pull/531))
```
Comment (jasonpaulos) [inline CHANGELOG.md:7]: I think this should go under Changed instead of Fixed

**Review by jasonpaulos — COMMENTED** [review]
I think `type_spec_is_assignable_to` should be used in `InnerTxnBuilder.MethodCall` as well instead of this equality check: https://github.com/algorand/pyteal/blob/221232a8855513d6435d0e6cc9eea6da273612ff/pyteal/ast/itxn.py#L462

**Comment by ahangsu** [comment]
`InnerTxnBuilder.MethodCall` ticket is #537, will file another PR for that

File: CHANGELOG.md (hunk 2 above)
Comment (ahangsu) [reply CHANGELOG.md]: change pushed

**Review by jasonpaulos — APPROVED** [review]
Thanks for pointing me to #537, I was unaware of that. Since a separate issue exists for inner txns, this looks good to me.

**Pull request merged**